packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -40,8 +43,8 @@ import {
 
 export function _publishCloudEventSend(
   context: Client,
-  event: CloudEvent,
   topicName: string,
+  event: CloudEvent,
```

Comment:
In this case, both the previous change and new change are flattened payload because their typespec is defined as 
```
  @post op PublishCloudEvent is Azure.Core.RpcOperation<{
    @doc("content type")
    @header("content-type")
    contentType: "application/cloudevents+json; charset=utf-8";

    @doc("Topic Name.")
    @path
    topicName: string;

    @doc("Single Cloud Event being published.")
    @body 
    event: {event: CloudEvent};
  }, PublishResult>;
```
where the body parameter is actually an anonymous model, and we are using model.name === "" to determine if there's an alias, anonymous model will also fit into the condition, so we also need to flatten the payload here, as of the order change is from this comment

Reply:
Also, please note in their latest typespec from main branch, they already removed the anonymous model https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/Azure.Messaging.EventGrid/main.tsp#L197